src/java.base/share/man/java.md

```diff
@@ -1085,6 +1085,23 @@ These `java` options control the runtime behavior of the Java HotSpot VM.
 `-XX:CRaCRestoreFrom=`*directory*
 :   Restores from the specified checkpoint image.
 
+    When using this option, there is no requirement to specify a main class and its
+    arguments because the restored execution is based on the ones specified before
+    the checkpoint. However, it is possible to pass a new main class and new
+    arguments. After a successful restoration, the new main method will be invoked
+    with the specified arguments in the thread that started the checkpoint, on top
+    of its existing call stack.
```

Comment:
I'd suggest start with positive example, something like:

You can pass a new "main" class and arguments with this option. Then, a method matching `main(String[])` will be invoked then with the provided arguments after all `afterRestore` methods complete. The method will invoked on the thread which initiated the checkpoint.